packages/core/src/runtime/RuntimeLogger.ts

```diff
@@ -0,0 +1,33 @@
+import type { Logger } from "../logging/logger.ts";
+
+export class RuntimeLogger {
+    private loggers: Logger[];
+
+    constructor(loggers: Logger[] = []) {
+        this.loggers = loggers;
+    }
+
+    private log(action: (logger: Logger) => Promise<unknown>): Promise<PromiseSettledResult<unknown>[]> {
+        return Promise.allSettled(this.loggers.map((x: Logger) => action(x)));
+    }
+
+    debug(log: string, ...rest: unknown[]): Promise<PromiseSettledResult<unknown>[]> {
+        return this.log((x: Logger) => x.debug(log, ...rest));
```

Comment:
pas besoin de typer les x: Logger quand tu passe une lambda direct a une fonction, il comprend que la fonction accepte une func qui recoit un logger.

return this.log(x => x.debug(log, ...rest));